dhcpv4/bsdp/bsdp.go

```diff
@@ -0,0 +1,315 @@
+// +build darwin
+
+package bsdp
+
+// Implements Apple's netboot protocol BSDP (Boot Service Discovery Protocol).
+// Canonical implementation is defined here:
+// http://opensource.apple.com/source/bootp/bootp-198.1/Documentation/BSDP.doc
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// BootImageID describes a boot image ID - whether it's an install image and
+// what kind of boot image (e.g. OS 9, macOS, hardware diagnostics)
+type BootImageID struct {
+	IsInstall bool
+	ImageType byte
+	Index     uint16
+}
+
+// ToBytes serializes a BootImageID to network-order bytes.
+func (b BootImageID) ToBytes() (bytes []byte) {
+	bytes = make([]byte, 4)
+	if b.IsInstall {
+		bytes[0] |= 0x80
+	}
+	bytes[0] |= b.ImageType
+	binary.BigEndian.PutUint16(bytes[2:], b.Index)
+	return
+}
+
+// BootImageIDFromBytes deserializes a collection of 4 bytes to a BootImageID.
+func BootImageIDFromBytes(bytes []byte) BootImageID {
+	return BootImageID{
+		IsInstall: bytes[0]&0x80 != 0,
+		ImageType: bytes[0] & 0x7f,
+		Index:     binary.BigEndian.Uint16(bytes[2:]),
+	}
+}
+
+// BootImage describes a boot image - contains the boot image ID and the name.
+type BootImage struct {
+	ID   BootImageID
+	Name string
+}
+
+// ToBytes converts a BootImage to a slice of bytes.
+func (b *BootImage) ToBytes() (bytes []byte) {
```

Comment:
nit: for consistency with the rest of the code can you return `bytes` as a non-named type?